mount/lookup_unix.go

```diff
@@ -20,34 +20,21 @@ package mount
 
 import (
 	"path/filepath"
-	"sort"
-	"strings"
 
+	"github.com/moby/sys/mountinfo"
 	"github.com/pkg/errors"
 )
 
 // Lookup returns the mount info corresponds to the path.
 func Lookup(dir string) (Info, error) {
 	dir = filepath.Clean(dir)
 
-	mounts, err := Self()
+	m, err := mountinfo.GetMounts(mountinfo.SingleEntryFilter(dir))
```

Comment:
Alas, this is wrong. Previously this function was capable of returning info for e.g. mount point `/mnt` given the argument of `/mnt/some/thing`. Currently it looks for `/mnt/some/thing`.

You need `mountinfo.ParentsFilter(dir)`, sort, and take the longest one -- same as in `getSourceMount` in moby/moby.

Reply:
I also think we can implement this in mountinfo itself, as it's been used from a few places.

Reply:
updated; PTAL